src/ui.c

```diff
@@ -1007,94 +1007,34 @@ static void DiskManagement(void)
-		UI_MENU_ACTION(CARTRIDGE_PHOENIX_8, CARTRIDGE_PHOENIX_8_DESC),
-		UI_MENU_ACTION(CARTRIDGE_BLIZZARD_16, CARTRIDGE_BLIZZARD_16_DESC),
-		UI_MENU_ACTION(CARTRIDGE_ATMAX_128, CARTRIDGE_ATMAX_128_DESC),
-		UI_MENU_ACTION(CARTRIDGE_ATMAX_1024, CARTRIDGE_ATMAX_1024_DESC),
-		UI_MENU_ACTION(CARTRIDGE_SDX_128, CARTRIDGE_SDX_128_DESC),
-		UI_MENU_ACTION(CARTRIDGE_OSS_8, CARTRIDGE_OSS_8_DESC),
-		UI_MENU_ACTION(CARTRIDGE_OSS_043M_16, CARTRIDGE_OSS_043M_16_DESC),
-		UI_MENU_ACTION(CARTRIDGE_BLIZZARD_4, CARTRIDGE_BLIZZARD_4_DESC),
-		UI_MENU_ACTION(CARTRIDGE_AST_32, CARTRIDGE_AST_32_DESC),
-		UI_MENU_ACTION(CARTRIDGE_ATRAX_SDX_64, CARTRIDGE_ATRAX_SDX_64_DESC),
-		UI_MENU_ACTION(CARTRIDGE_ATRAX_SDX_128, CARTRIDGE_ATRAX_SDX_128_DESC),
-		UI_MENU_ACTION(CARTRIDGE_TURBOSOFT_64, CARTRIDGE_TURBOSOFT_64_DESC),
-		UI_MENU_ACTION(CARTRIDGE_TURBOSOFT_128, CARTRIDGE_TURBOSOFT_128_DESC),
-		UI_MENU_ACTION(CARTRIDGE_ULTRACART_32, CARTRIDGE_ULTRACART_32_DESC),
-		UI_MENU_ACTION(CARTRIDGE_LOW_BANK_8, CARTRIDGE_LOW_BANK_8_DESC),
-		UI_MENU_ACTION(CARTRIDGE_SIC_128, CARTRIDGE_SIC_128_DESC),
-		UI_MENU_ACTION(CARTRIDGE_SIC_256, CARTRIDGE_SIC_256_DESC),
-		UI_MENU_ACTION(CARTRIDGE_SIC_512, CARTRIDGE_SIC_512_DESC),
-		UI_MENU_ACTION(CARTRIDGE_STD_2, CARTRIDGE_STD_2_DESC),
-		UI_MENU_ACTION(CARTRIDGE_STD_4, CARTRIDGE_STD_4_DESC),
-		UI_MENU_ACTION(CARTRIDGE_RIGHT_4, CARTRIDGE_RIGHT_4_DESC),
-		UI_MENU_ACTION(CARTRIDGE_BLIZZARD_32, CARTRIDGE_BLIZZARD_32_DESC),
-		UI_MENU_ACTION(CARTRIDGE_MEGAMAX_2048, CARTRIDGE_MEGAMAX_2048_DESC),
-		UI_MENU_ACTION(CARTRIDGE_THECART_128M, CARTRIDGE_THECART_128M_DESC),
-		UI_MENU_ACTION(CARTRIDGE_MEGA_4096, CARTRIDGE_MEGA_4096_DESC),
-		UI_MENU_ACTION(CARTRIDGE_MEGA_2048, CARTRIDGE_MEGA_2048_DESC),
-		UI_MENU_ACTION(CARTRIDGE_THECART_32M, CARTRIDGE_THECART_32M_DESC),
-		UI_MENU_ACTION(CARTRIDGE_THECART_64M, CARTRIDGE_THECART_64M_DESC),
-		UI_MENU_ACTION(CARTRIDGE_XEGS_8F_64, CARTRIDGE_XEGS_8F_64_DESC),
-		UI_MENU_ACTION(CARTRIDGE_ATRAX_128, CARTRIDGE_ATRAX_128_DESC),
-		UI_MENU_ACTION(CARTRIDGE_ADAWLIAH_32, CARTRIDGE_ADAWLIAH_32_DESC),
-		UI_MENU_ACTION(CARTRIDGE_ADAWLIAH_64, CARTRIDGE_ADAWLIAH_64_DESC),
-		UI_MENU_END
-	};
-
-	int i;
+	/* menu_array must be terminated with UI_MENU_END */
+	UI_tMenuItem menu_array[CARTRIDGE_LAST_SUPPORTED+1]; 
+	int cart_entry, menu_entry;
 	int option = 0;
+	UWORD flag;
 
 	UI_driver->fInit();
 
-	for (i = 1; i <= CARTRIDGE_LAST_SUPPORTED; i++)
-		if (CARTRIDGE_kb[i] == k) {
+	for (cart_entry = 1; cart_entry <= CARTRIDGE_LAST_SUPPORTED;
+	     cart_entry++) {
+		/* N.B.: CARTRIDGES[] is offset by 1 from menu_array[]
+		   due to CARTRIDGE_NONE entry. */
+		menu_entry = cart_entry - 1;
+
+		if (CARTRIDGES[cart_entry].kb == k) {
 			if (option == 0)
-				option = i;
-			menu_array[i - 1].flags = UI_ITEM_ACTION;
+				option = cart_entry;
+			flag = UI_ITEM_ACTION;
 		}
 		else
-			menu_array[i - 1].flags = UI_ITEM_HIDDEN;
+			flag = UI_ITEM_HIDDEN;
```

Comment:
As you are dynamically creating the menu_array, you could simply include only the cartridge types actually supported, and skip the other ones, something like this:
```
	for (i = 1, menu_entry = 0; i <= CARTRIDGE_LAST_SUPPORTED; i++)
		if (CARTRIDGES[i].kb == k) {
			UI_MAKE_MENU_ACTION(menu_array[menu_entry], UI_ITEM_ACTION, i, CARTRIDGES[i].description);
			menu_entry++;
		}
	if(menu_entry == 0)
		return CARTRIDGE_NONE;
	UI_MAKE_END(menu_array[menu_entry]);
```